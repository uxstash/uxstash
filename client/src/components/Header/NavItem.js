import React from "react";
import { NavLink, withRouter } from "react-router-dom";

export const NavItem = props => {
  return (
    <li className={props.to === props.location.pathname ? "active" : ""}>
      <NavLink to={props.to}>
        {props.title}
        <small>{props.description}</small>
      </NavLink>
    </li>
  );
};

export default withRouter(NavItem);
