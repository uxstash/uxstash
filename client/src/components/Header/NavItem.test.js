import React from "react";
import { shallow, mount } from "enzyme";
import { MemoryRouter, Route } from "react-router-dom";

import NavItem from "./NavItem";

describe("NavItem", () => {
  it("should render correctly with props", () => {
    const title = "test title";
    const path = "/testpath";
    const description = "test description";

    const component = mount(
      <MemoryRouter initialEntries={[path]}>
        <Route
          render={props => (
            <NavItem
              {...props}
              to={path}
              title={title}
              description={description}
            />
          )}
        />
      </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
  });
});

describe("NavItem", () => {
  it("should render correctly with no props", () => {
    const component = mount(
      <MemoryRouter initialEntries={["/strategy"]}>
        <Route render={props => <NavItem />} />
      </MemoryRouter>
    );
    console.log(component.debug());
    expect(component).toMatchSnapshot();
  });
});
