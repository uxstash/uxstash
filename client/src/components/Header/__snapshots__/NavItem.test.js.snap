// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem should render correctly with no props 1`] = `
<li
  className=""
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/"
  >
    <Route
      path="\\\\/"
    >
      <Link
        aria-current="page"
        className="active"
        replace={false}
        style={Object {}}
        to="/"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <small />
        </a>
      </Link>
    </Route>
  </NavLink>
</li>
`;

exports[`NavItem should render correctly with props 1`] = `
<li
  className="active"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/testpath"
  >
    <Route
      path="\\\\/testpath"
    >
      <Link
        aria-current="page"
        className="active"
        replace={false}
        style={Object {}}
        to="/testpath"
      >
        <a
          aria-current="page"
          className="active"
          href="/testpath"
          onClick={[Function]}
          style={Object {}}
        >
          test title
          <small />
        </a>
      </Link>
    </Route>
  </NavLink>
</li>
`;
